From 0a5df8cc725f6781bf0d8b9478af22357e5f10b4 Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Tue, 21 Jun 2011 17:03:53 +0300
Subject: [PATCH] wpa_gui: make the networkconfig dialog scrollable

Makes the networkconfig dialog scrollable to allow using it on screens
with low resolution in the "y" axis

Signed-off-by: Eyal Reizer <eyalr@ti.com>
---
 networkconfig.ui |  817 ++++++++++++++-------------
 1 files changed, 409 insertions(+), 408 deletions(-)

diff --git a/networkconfig.ui b/networkconfig.ui
index 217a8ff..d90bd64 100644
--- a/networkconfig.ui
+++ b/networkconfig.ui
@@ -1,434 +1,435 @@
-<ui version="4.0" >
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>NetworkConfig</class>
- <widget class="QDialog" name="NetworkConfig" >
-  <property name="geometry" >
+ <widget class="QDialog" name="NetworkConfig">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>410</width>
-    <height>534</height>
+    <width>430</width>
+    <height>613</height>
    </rect>
   </property>
-  <property name="windowTitle" >
+  <property name="windowTitle">
    <string>NetworkConfig</string>
   </property>
-  <layout class="QGridLayout" >
-   <item row="1" column="3" >
-    <widget class="QPushButton" name="cancelButton" >
-     <property name="text" >
-      <string>Cancel</string>
+  <layout class="QGridLayout">
+   <item row="0" column="0">
+    <widget class="QScrollArea" name="scrollArea">
+     <property name="widgetResizable">
+      <bool>true</bool>
      </property>
-    </widget>
-   </item>
-   <item row="0" column="0" colspan="4" >
-    <widget class="QFrame" name="frame9" >
-     <property name="frameShape" >
-      <enum>QFrame::NoFrame</enum>
-     </property>
-     <property name="frameShadow" >
-      <enum>QFrame::Plain</enum>
-     </property>
-     <layout class="QGridLayout" >
-      <item row="0" column="0" >
-       <widget class="QLabel" name="ssidLabel" >
-        <property name="text" >
-         <string>SSID</string>
-        </property>
-       </widget>
-      </item>
-      <item row="0" column="1" >
-       <widget class="QLineEdit" name="ssidEdit" >
-        <property name="toolTip" >
-         <string>Network name (Service Set IDentifier)</string>
-        </property>
-        <property name="text" >
-         <string/>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="0" >
-       <widget class="QLabel" name="authLabel" >
-        <property name="text" >
-         <string>Authentication</string>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="1" >
-       <widget class="QComboBox" name="authSelect" >
-        <item>
-         <property name="text" >
-          <string>Plaintext (open / no authentication)</string>
-         </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>Static WEP (no authentication)</string>
-         </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>Static WEP (Shared Key authentication)</string>
+     <widget class="QWidget" name="scrollAreaWidgetContents">
+      <property name="geometry">
+       <rect>
+        <x>0</x>
+        <y>0</y>
+        <width>430</width>
+        <height>613</height>
+       </rect>
+      </property>
+      <layout class="QGridLayout" name="gridLayout">
+       <item row="0" column="0" colspan="4">
+        <widget class="QFrame" name="frame9">
+         <property name="frameShape">
+          <enum>QFrame::NoFrame</enum>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>IEEE 802.1X</string>
+         <property name="frameShadow">
+          <enum>QFrame::Plain</enum>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>WPA-Personal (PSK)</string>
+         <layout class="QGridLayout" name="_2">
+          <item row="0" column="0">
+           <widget class="QLabel" name="ssidLabel">
+            <property name="text">
+             <string>SSID</string>
+            </property>
+           </widget>
+          </item>
+          <item row="0" column="1">
+           <widget class="QLineEdit" name="ssidEdit">
+            <property name="toolTip">
+             <string>Network name (Service Set IDentifier)</string>
+            </property>
+            <property name="text">
+             <string/>
+            </property>
+           </widget>
+          </item>
+          <item row="1" column="0">
+           <widget class="QLabel" name="authLabel">
+            <property name="text">
+             <string>Authentication</string>
+            </property>
+           </widget>
+          </item>
+          <item row="1" column="1">
+           <widget class="QComboBox" name="authSelect">
+            <item>
+             <property name="text">
+              <string>Plaintext (open / no authentication)</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>Static WEP (no authentication)</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>Static WEP (Shared Key authentication)</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>IEEE 802.1X</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>WPA-Personal (PSK)</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>WPA-Enterprise (EAP)</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>WPA2-Personal (PSK)</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>WPA2-Enterprise (EAP)</string>
+             </property>
+            </item>
+           </widget>
+          </item>
+          <item row="2" column="0">
+           <widget class="QLabel" name="encrLabel">
+            <property name="text">
+             <string>Encryption</string>
+            </property>
+           </widget>
+          </item>
+          <item row="2" column="1">
+           <widget class="QComboBox" name="encrSelect">
+            <item>
+             <property name="text">
+              <string>None</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>WEP</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>TKIP</string>
+             </property>
+            </item>
+            <item>
+             <property name="text">
+              <string>CCMP</string>
+             </property>
+            </item>
+           </widget>
+          </item>
+          <item row="3" column="0">
+           <widget class="QLabel" name="pskLabel">
+            <property name="text">
+             <string>PSK</string>
+            </property>
+           </widget>
+          </item>
+          <item row="3" column="1">
+           <widget class="QLineEdit" name="pskEdit">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
+            <property name="toolTip">
+             <string>WPA/WPA2 pre-shared key or passphrase</string>
+            </property>
+            <property name="whatsThis">
+             <string/>
+            </property>
+            <property name="echoMode">
+             <enum>QLineEdit::Password</enum>
+            </property>
+           </widget>
+          </item>
+          <item row="4" column="0">
+           <widget class="QLabel" name="eapLabel">
+            <property name="text">
+             <string>EAP method</string>
+            </property>
+           </widget>
+          </item>
+          <item row="4" column="1">
+           <widget class="QComboBox" name="eapSelect">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
+           </widget>
+          </item>
+          <item row="5" column="0">
+           <widget class="QLabel" name="identityLabel">
+            <property name="text">
+             <string>Identity</string>
+            </property>
+           </widget>
+          </item>
+          <item row="5" column="1">
+           <widget class="QLineEdit" name="identityEdit">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
+            <property name="toolTip">
+             <string>Username/Identity for EAP methods</string>
+            </property>
+           </widget>
+          </item>
+          <item row="6" column="0">
+           <widget class="QLabel" name="passwordLabel">
+            <property name="text">
+             <string>Password</string>
+            </property>
+           </widget>
+          </item>
+          <item row="6" column="1">
+           <widget class="QLineEdit" name="passwordEdit">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
+            <property name="toolTip">
+             <string>Password for EAP methods</string>
+            </property>
+            <property name="echoMode">
+             <enum>QLineEdit::Password</enum>
+            </property>
+           </widget>
+          </item>
+          <item row="7" column="0">
+           <widget class="QLabel" name="cacertLabel">
+            <property name="text">
+             <string>CA certificate</string>
+            </property>
+           </widget>
+          </item>
+          <item row="7" column="1">
+           <widget class="QLineEdit" name="cacertEdit">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
+           </widget>
+          </item>
+          <item row="8" column="0" colspan="2">
+           <widget class="QGroupBox" name="wepBox">
+            <property name="enabled">
+             <bool>true</bool>
+            </property>
+            <property name="title">
+             <string>WEP keys</string>
+            </property>
+            <layout class="QGridLayout" name="_3">
+             <item row="0" column="0">
+              <widget class="QRadioButton" name="wep0Radio">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+               <property name="text">
+                <string>key 0</string>
+               </property>
+              </widget>
+             </item>
+             <item row="1" column="0">
+              <widget class="QRadioButton" name="wep1Radio">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+               <property name="text">
+                <string>key 1</string>
+               </property>
+              </widget>
+             </item>
+             <item row="3" column="0">
+              <widget class="QRadioButton" name="wep3Radio">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+               <property name="text">
+                <string>key 3</string>
+               </property>
+              </widget>
+             </item>
+             <item row="2" column="0">
+              <widget class="QRadioButton" name="wep2Radio">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+               <property name="text">
+                <string>key 2</string>
+               </property>
+              </widget>
+             </item>
+             <item row="0" column="1">
+              <widget class="QLineEdit" name="wep0Edit">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+              </widget>
+             </item>
+             <item row="1" column="1">
+              <widget class="QLineEdit" name="wep1Edit">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+              </widget>
+             </item>
+             <item row="2" column="1">
+              <widget class="QLineEdit" name="wep2Edit">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+              </widget>
+             </item>
+             <item row="3" column="1">
+              <widget class="QLineEdit" name="wep3Edit">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+              </widget>
+             </item>
+            </layout>
+           </widget>
+          </item>
+          <item row="9" column="0" colspan="2">
+           <widget class="QGroupBox" name="optionalSettingsBox">
+            <property name="enabled">
+             <bool>true</bool>
+            </property>
+            <property name="title">
+             <string>Optional Settings</string>
+            </property>
+            <layout class="QGridLayout" name="_4">
+             <item row="0" column="1">
+              <widget class="QLineEdit" name="idstrEdit">
+               <property name="toolTip">
+                <string>Network Identification String</string>
+               </property>
+              </widget>
+             </item>
+             <item row="0" column="3">
+              <widget class="QSpinBox" name="prioritySpinBox">
+               <property name="toolTip">
+                <string>Network Priority</string>
+               </property>
+               <property name="maximum">
+                <number>10000</number>
+               </property>
+               <property name="singleStep">
+                <number>10</number>
+               </property>
+              </widget>
+             </item>
+             <item row="0" column="0">
+              <widget class="QLabel" name="idstrLabel">
+               <property name="text">
+                <string>IDString</string>
+               </property>
+              </widget>
+             </item>
+             <item row="0" column="2">
+              <widget class="QLabel" name="priorityLabel">
+               <property name="text">
+                <string>Priority</string>
+               </property>
+              </widget>
+             </item>
+             <item row="1" column="0">
+              <widget class="QLabel" name="phase2Label">
+               <property name="text">
+                <string>Inner auth</string>
+               </property>
+              </widget>
+             </item>
+             <item row="1" column="1">
+              <widget class="QComboBox" name="phase2Select">
+               <property name="enabled">
+                <bool>false</bool>
+               </property>
+              </widget>
+             </item>
+            </layout>
+           </widget>
+          </item>
+         </layout>
+        </widget>
+       </item>
+       <item row="1" column="0">
+        <spacer name="spacer">
+         <property name="orientation">
+          <enum>Qt::Vertical</enum>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>WPA-Enterprise (EAP)</string>
+         <property name="sizeHint" stdset="0">
+          <size>
+           <width>20</width>
+           <height>24</height>
+          </size>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>WPA2-Personal (PSK)</string>
+        </spacer>
+       </item>
+       <item row="1" column="1">
+        <widget class="QPushButton" name="useWpsButton">
+         <property name="enabled">
+          <bool>false</bool>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>WPA2-Enterprise (EAP)</string>
+         <property name="text">
+          <string>WPS</string>
          </property>
-        </item>
-       </widget>
-      </item>
-      <item row="2" column="0" >
-       <widget class="QLabel" name="encrLabel" >
-        <property name="text" >
-         <string>Encryption</string>
-        </property>
-       </widget>
-      </item>
-      <item row="2" column="1" >
-       <widget class="QComboBox" name="encrSelect" >
-        <item>
-         <property name="text" >
-          <string>None</string>
+        </widget>
+       </item>
+       <item row="1" column="2">
+        <widget class="QPushButton" name="addButton">
+         <property name="text">
+          <string>Add</string>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>WEP</string>
+        </widget>
+       </item>
+       <item row="1" column="3">
+        <widget class="QPushButton" name="removeButton">
+         <property name="enabled">
+          <bool>false</bool>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>TKIP</string>
+         <property name="text">
+          <string>Remove</string>
          </property>
-        </item>
-        <item>
-         <property name="text" >
-          <string>CCMP</string>
-         </property>
-        </item>
-       </widget>
-      </item>
-      <item row="3" column="0" >
-       <widget class="QLabel" name="pskLabel" >
-        <property name="text" >
-         <string>PSK</string>
-        </property>
-       </widget>
-      </item>
-      <item row="3" column="1" >
-       <widget class="QLineEdit" name="pskEdit" >
-        <property name="enabled" >
-         <bool>false</bool>
-        </property>
-        <property name="toolTip" >
-         <string>WPA/WPA2 pre-shared key or passphrase</string>
-        </property>
-        <property name="whatsThis" >
-         <string/>
-        </property>
-        <property name="echoMode" >
-         <enum>QLineEdit::Password</enum>
-        </property>
-       </widget>
-      </item>
-      <item row="4" column="0" >
-       <widget class="QLabel" name="eapLabel" >
-        <property name="text" >
-         <string>EAP method</string>
-        </property>
-       </widget>
-      </item>
-      <item row="4" column="1" >
-       <widget class="QComboBox" name="eapSelect" >
-        <property name="enabled" >
-         <bool>false</bool>
-        </property>
-       </widget>
-      </item>
-      <item row="5" column="0" >
-       <widget class="QLabel" name="identityLabel" >
-        <property name="text" >
-         <string>Identity</string>
-        </property>
-       </widget>
-      </item>
-      <item row="5" column="1" >
-       <widget class="QLineEdit" name="identityEdit" >
-        <property name="enabled" >
-         <bool>false</bool>
-        </property>
-        <property name="toolTip" >
-         <string>Username/Identity for EAP methods</string>
-        </property>
-       </widget>
-      </item>
-      <item row="6" column="0" >
-       <widget class="QLabel" name="passwordLabel" >
-        <property name="text" >
-         <string>Password</string>
-        </property>
-       </widget>
-      </item>
-      <item row="6" column="1" >
-       <widget class="QLineEdit" name="passwordEdit" >
-        <property name="enabled" >
-         <bool>false</bool>
-        </property>
-        <property name="toolTip" >
-         <string>Password for EAP methods</string>
-        </property>
-        <property name="echoMode" >
-         <enum>QLineEdit::Password</enum>
-        </property>
-       </widget>
-      </item>
-      <item row="7" column="0" >
-       <widget class="QLabel" name="cacertLabel" >
-        <property name="text" >
-         <string>CA certificate</string>
-        </property>
-       </widget>
-      </item>
-      <item row="7" column="1" >
-       <widget class="QLineEdit" name="cacertEdit" >
-        <property name="enabled" >
-         <bool>false</bool>
-        </property>
-       </widget>
-      </item>
-      <item row="8" column="0" colspan="2" >
-       <widget class="QGroupBox" name="wepBox" >
-        <property name="enabled" >
-         <bool>true</bool>
-        </property>
-        <property name="title" >
-         <string>WEP keys</string>
-        </property>
-        <layout class="QGridLayout" >
-         <item row="0" column="0" >
-          <widget class="QRadioButton" name="wep0Radio" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-           <property name="text" >
-            <string>key 0</string>
-           </property>
-          </widget>
-         </item>
-         <item row="1" column="0" >
-          <widget class="QRadioButton" name="wep1Radio" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-           <property name="text" >
-            <string>key 1</string>
-           </property>
-          </widget>
-         </item>
-         <item row="3" column="0" >
-          <widget class="QRadioButton" name="wep3Radio" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-           <property name="text" >
-            <string>key 3</string>
-           </property>
-          </widget>
-         </item>
-         <item row="2" column="0" >
-          <widget class="QRadioButton" name="wep2Radio" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-           <property name="text" >
-            <string>key 2</string>
-           </property>
-          </widget>
-         </item>
-         <item row="0" column="1" >
-          <widget class="QLineEdit" name="wep0Edit" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-          </widget>
-         </item>
-         <item row="1" column="1" >
-          <widget class="QLineEdit" name="wep1Edit" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-          </widget>
-         </item>
-         <item row="2" column="1" >
-          <widget class="QLineEdit" name="wep2Edit" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-          </widget>
-         </item>
-         <item row="3" column="1" >
-          <widget class="QLineEdit" name="wep3Edit" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-          </widget>
-         </item>
-        </layout>
-       </widget>
-      </item>
-      <item row="9" column="0" colspan="2" >
-       <widget class="QGroupBox" name="optionalSettingsBox" >
-        <property name="enabled" >
-         <bool>true</bool>
-        </property>
-        <property name="title" >
-         <string>Optional Settings</string>
-        </property>
-        <layout class="QGridLayout" >
-         <item row="0" column="1" >
-          <widget class="QLineEdit" name="idstrEdit" >
-           <property name="toolTip" >
-            <string>Network Identification String</string>
-           </property>
-          </widget>
-         </item>
-         <item row="0" column="3" >
-          <widget class="QSpinBox" name="prioritySpinBox" >
-           <property name="toolTip" >
-            <string>Network Priority</string>
-           </property>
-           <property name="maximum" >
-            <number>10000</number>
-           </property>
-           <property name="singleStep" >
-            <number>10</number>
-           </property>
-          </widget>
-         </item>
-         <item row="0" column="0" >
-          <widget class="QLabel" name="idstrLabel" >
-           <property name="text" >
-            <string>IDString</string>
-           </property>
-          </widget>
-         </item>
-         <item row="0" column="2" >
-          <widget class="QLabel" name="priorityLabel" >
-           <property name="text" >
-            <string>Priority</string>
-           </property>
-          </widget>
-         </item>
-         <item row="1" column="0" >
-          <widget class="QLabel" name="phase2Label" >
-           <property name="text" >
-            <string>Inner auth</string>
-           </property>
-          </widget>
-         </item>
-         <item row="1" column="1" >
-          <widget class="QComboBox" name="phase2Select" >
-           <property name="enabled" >
-            <bool>false</bool>
-           </property>
-          </widget>
-         </item>
-        </layout>
-       </widget>
-      </item>
-     </layout>
-    </widget>
-   </item>
-   <item row="1" column="2" >
-    <widget class="QPushButton" name="addButton" >
-     <property name="text" >
-      <string>Add</string>
-     </property>
-    </widget>
-   </item>
-   <item row="1" column="3" >
-    <widget class="QPushButton" name="removeButton" >
-     <property name="enabled" >
-      <bool>false</bool>
-     </property>
-     <property name="text" >
-      <string>Remove</string>
-     </property>
-    </widget>
-   </item>
-   <item row="1" column="0" >
-    <spacer>
-     <property name="orientation" >
-      <enum>Qt::Vertical</enum>
-     </property>
-     <property name="sizeHint" >
-      <size>
-       <width>20</width>
-       <height>40</height>
-      </size>
-     </property>
-    </spacer>
-   </item>
-   <item row="1" column="1" >
-    <widget class="QPushButton" name="useWpsButton" >
-     <property name="enabled" >
-      <bool>false</bool>
-     </property>
-     <property name="text" >
-      <string>WPS</string>
-     </property>
+        </widget>
+       </item>
+      </layout>
+      <widget class="QPushButton" name="cancelButton">
+       <property name="geometry">
+        <rect>
+         <x>454</x>
+         <y>798</y>
+         <width>98</width>
+         <height>27</height>
+        </rect>
+       </property>
+       <property name="text">
+        <string>Cancel</string>
+       </property>
+      </widget>
+     </widget>
     </widget>
    </item>
   </layout>
  </widget>
- <layoutdefault spacing="6" margin="11" />
- <pixmapfunction></pixmapfunction>
- <tabstops>
-  <tabstop>ssidEdit</tabstop>
-  <tabstop>authSelect</tabstop>
-  <tabstop>encrSelect</tabstop>
-  <tabstop>pskEdit</tabstop>
-  <tabstop>eapSelect</tabstop>
-  <tabstop>identityEdit</tabstop>
-  <tabstop>passwordEdit</tabstop>
-  <tabstop>cacertEdit</tabstop>
-  <tabstop>wep0Radio</tabstop>
-  <tabstop>wep0Edit</tabstop>
-  <tabstop>wep1Radio</tabstop>
-  <tabstop>wep1Edit</tabstop>
-  <tabstop>wep2Radio</tabstop>
-  <tabstop>wep2Edit</tabstop>
-  <tabstop>wep3Radio</tabstop>
-  <tabstop>wep3Edit</tabstop>
-  <tabstop>idstrEdit</tabstop>
-  <tabstop>prioritySpinBox</tabstop>
-  <tabstop>phase2Select</tabstop>
-  <tabstop>addButton</tabstop>
-  <tabstop>removeButton</tabstop>
-  <tabstop>cancelButton</tabstop>
- </tabstops>
+ <layoutdefault spacing="6" margin="11"/>
  <includes>
-  <include location="global" >qtreewidget.h</include>
+  <include location="global">qtreewidget.h</include>
  </includes>
  <resources/>
  <connections/>
--
1.7.0.4

